src/neo/SmartContract/ApplicationEngine.Contract.cs

```diff
@@ -50,7 +50,18 @@ protected internal ContractState CreateContract(byte[] script, byte[] manifest)
             if (!contract.Manifest.IsValid(hash)) throw new InvalidOperationException($"Invalid Manifest Hash: {hash}");
 
             Snapshot.Contracts.Add(hash, contract);
-            return contract;
+
+            // We should push it onto the caller's stack.
+
+            Push(Convert(contract));
+
+            // Execute _deploy
+
+            ContractMethodDescriptor md = contract.Manifest.Abi.GetMethod("_deploy");
+            if (md != null)
+                CallContractInternal(contract, md, new Array(ReferenceCounter) { false }, CallFlags.All, false);
```

Comment:
If you set `NeedCheckReturnValue` to `false`, the method (`_deploy`) can push any values onto the stack.

Reply:
How do you think that we can remove the null value?

Reply:
@erikzhang please take a look again, it could solve the problem